lib/api/core/validation/types/anything.js

```diff
@@ -0,0 +1,29 @@
+var
+  BaseConstructor = require('../baseType');
+
+/**
+ * @constructor
+ */
+function AnythingType () {
+  this.typeName = 'anything';
+  this.allowChildren = false;
+  this.allowedTypeOptions = [];
+}
+
+AnythingType.prototype = new BaseConstructor();
```

Comment:
nitpicking: `Anything.constructor` is not set to `Anything`. Also applies to all other types.

```js
var o = new AnythingType(); 
console.log(o.constructor);
[Function: BaseType]
```

Whe can use [`util.inherits`](https://nodejs.org/docs/latest/api/util.html#util_util_inherits_constructor_superconstructor) here.